tests/components/onewire/test_entity_sysbus.py

```diff
@@ -92,30 +128,46 @@
 async def test_onewiredirect_setup_valid_device(hass, device_id):
     """Test that sysbus config entry works correctly."""
     entity_registry = mock_registry(hass)
+    device_registry = mock_device_registry(hass)
+
+    mock_device_sensor = MOCK_DEVICE_SENSORS[device_id]
 
+    glob_result = [f"/{DEFAULT_SYSBUS_MOUNT_DIR}/{device_id}"]
     read_side_effect = []
-    expected_sensors = MOCK_DEVICE_SENSORS[device_id]["sensors"]
+    expected_sensors = mock_device_sensor["sensors"]
     for expected_sensor in expected_sensors:
         read_side_effect.append(expected_sensor["injected_value"])
 
+    # Ensure enough read side effect
+    read_side_effect.extend([FileNotFoundError("Missing injected value")] * 20)
+
     with patch(
-        "homeassistant.components.onewire.sensor.Pi1Wire"
-    ) as mock_pi1wire, patch("pi1wire.OneWire") as mock_owsensor:
-        type(mock_owsensor).mac_address = PropertyMock(
-            return_value=device_id.replace("-", "")
-        )
-        mock_owsensor.get_temperature.side_effect = read_side_effect
-        mock_pi1wire.return_value.find_all_sensors.return_value = [mock_owsensor]
+        "homeassistant.components.onewire.sensor.os.path.isdir", return_value=True
+    ), patch("pi1wire._finder.glob.glob", return_value=glob_result,), patch(
+        "pi1wire.OneWire.get_temperature",
+        side_effect=read_side_effect,
+    ):
         assert await async_setup_component(hass, SENSOR_DOMAIN, MOCK_CONFIG)
         await hass.async_block_till_done()
 
     assert len(entity_registry.entities) == len(expected_sensors)
 
+    if len(expected_sensors) > 0:
+        device_info = mock_device_sensor["device_info"]
+        assert len(device_registry.devices) == 1
+        registry_entry = device_registry.async_get_device({(DOMAIN, device_id)}, set())
+        assert registry_entry is not None
+        assert registry_entry.identifiers == {(DOMAIN, device_id)}
+        assert registry_entry.manufacturer == device_info["manufacturer"]
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)